modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/transaction/Branch.java

```diff
@@ -77,4 +109,53 @@ public void rollback() {
     public Xid getXid() {
         return xid;
     }
+
+    public void setState(State state) {
+        this.state = state;
+    }
+
+    public State getState() {
+        return state;
+    }
+
+    /**
+     * Associate a session to current branch.
+     *
+     * @param sessionId session identifier of the session
+     */
+    public void associateSession(int sessionId) {
+        associatedSessions.put(sessionId, State.ACTIVE);
+    }
+
+    /**
+     * Resume a session if it is suspended
+     *
+     * @param sessionId session identifier of the session
+     */
+    public void resumeSession(int sessionId) {
+        if (associatedSessions.containsKey(sessionId) && associatedSessions.get(sessionId) == State.SUSPENDED) {
+            associatedSessions.put(sessionId, State.ACTIVE);
+        }
```

Comment:
Don't we need to repsond with an error if we are trying to resume a session that we did not suspend?